drivers/serial/uart_tt_virt.c

```diff
@@ -165,33 +163,9 @@ static void uart_tt_virt_irq_callback_set(const struct device *dev,
 	data->irq_cb_udata = user_data;
 }
 
-static void uart_tt_virt_irq_err_disable(const struct device *dev)
+static void uart_tt_virt_irq_handler(struct k_timer *timer)
 {
-	struct uart_tt_virt_data *const data = dev->data;
-
-	K_SPINLOCK(&data->err_lock) {
-		data->err_irq_en = false;
-	}
-}
-
-static void uart_tt_virt_irq_err_enable(const struct device *dev)
-{
-	bool submit;
-	struct uart_tt_virt_data *const data = dev->data;
-
-	K_SPINLOCK(&data->err_lock) {
-		data->err_irq_en = true;
-		submit = !!data->err_flags;
-	}
-
-	if (submit) {
-		(void)k_work_submit(&data->irq_work);
-	}
-}
```

Comment:
What's the reason for removing `err_disable()` and `err_enable()`?

I think they should probably be reworked. The idea was that this would be used against upstream test suites, and I think one of them required this functionality.

Reply:
I didn't love the requirement to nest two spinlocks to check if the interrupt could be disabled, and I didn't see these APIs being used. I've added them back though and reworked them